lib/smb2/file.rb

```diff
@@ -34,21 +48,28 @@ def initialize(tree:, create_response:)
   #
   # @return [Smb2::Packet::CloseResponse]
   def close
-    packet = Smb2::Packet::CloseRequest.new do |request|
-      request.file_id = self.create_response.file_id
-    end
+    packet = Smb2::Packet::CloseRequest.new(
+      file_id: self.create_response.file_id
+    )
 
     response = tree.send_recv(packet)
 
     Smb2::Packet::CloseResponse.new(response)
   end
 
+  # Whether the current read/write pointer is at the end of the file
+  #
+  # @return [Boolean]
+  def eof?
+    (@last_read_response && @last_read_response.header.nt_status == STATUS_END_OF_FILE) || pos == size
+  end
+
   # @return [String]
   def inspect
-    "#<Smb2::File file-id=#{create_response.file_id.unpack("H*").first} >"
+    "#<Smb2::File:#{filename} file-id=#{create_response.file_id.unpack("H*").first} >"
```

Comment:
This is a useful change, but wouldn't it be more useful to have the full UNC path since this won't include which share you're attached to since it doesn't include the tree.share?